### PR TITLE
Support options for GetCodeownersErrors

### LINKED ISSUE #2952: Supporting ref parameter for GetCodeownersErrors
It seems the `RepositoriesService.GetCodeownersErrors` method added in #2408 is missing support for the `ref` parameter. This is necessary to check the version of the file on specific branch of the repo.

### PR DESCRIPTION
**Pull request by lachlancooper**
Fixes #2952

### CODE REVIEW

File: github/repos_codeowners_test.go [hunk 1]
```diff
@@ -62,12 +62,74 @@ func TestRepositoriesService_GetCodeownersErrors(t *testing.T) {
...
+	})
+}
+
+func TestRepositoriesService_GetCodeownersErrors_specificRef(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/codeowners/errors", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeV3)
```
Comment (gmlewis) [inline github/repos_codeowners_test.go:84]: Please add `testFormValues` to this test.
For example, please see: https://github.com/google/go-github/blob/master/github/code-scanning_test.go#L139

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @lachlancooper !

File: github/repos_codeowners_test.go (hunk 1 above)
Comment (lachlancooper) [reply github/repos_codeowners_test.go:84]: Thanks, done!

**Review by gmlewis — APPROVED** [review]
Thank you, @lachlancooper !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by exageraldo — APPROVED** [review]

**Comment by exageraldo** [comment]
LGTM 👍🏻

**Pull request merged**